diff --git a/package/thingino-uhttpd/thingino-uhttpd.mk b/package/thingino-uhttpd/thingino-uhttpd.mk
--- a/package/thingino-uhttpd/thingino-uhttpd.mk
+++ b/package/thingino-uhttpd/thingino-uhttpd.mk
@@ -3,7 +3,7 @@ THINGINO_UHTTPD_SITE = https://git.openwrt.org/project/uhttpd.git
 THINGINO_UHTTPD_SITE_METHOD = git
 THINGINO_UHTTPD_LICENSE = ISC
 THINGINO_UHTTPD_LICENSE_FILES = uhttpd.h
-THINGINO_UHTTPD_DEPENDENCIES = thingino-libubox thingino-jct
+THINGINO_UHTTPD_DEPENDENCIES = thingino-libubox
 THINGINO_UHTTPD_CONF_OPTS += -DCMAKE_BUILD_TYPE=Debug
 
 ifeq ($(BR2_PACKAGE_LIBXCRYPT),y)
-- 
2.50.1
