diff --git a/package/exfat-nofuse/exfat-nofuse.mk b/package/exfat-nofuse/exfat-nofuse.mk
index 4a7cf5a7..7ee7a651 100644
--- a/package/exfat-nofuse/exfat-nofuse.mk
+++ b/package/exfat-nofuse/exfat-nofuse.mk
@@ -11,7 +11,7 @@ define EXFAT_NOFUSE_BUILD_CMDS
 endef
 
 define EXFAT_NOFUSE_INSTALL_TARGET_CMDS
-	krel="$$( $(MAKE) -s -C $(LINUX_DIR) kernelrelease 2>/dev/null )"; \
+	krel="$$( $(MAKE) --no-print-directory -s -C $(LINUX_DIR) kernelrelease 2>/dev/null )"; \
 	if [ -z "$$krel" ]; then krel="$(LINUX_VERSION_PROBED)"; fi; \
 	for root in "$(TARGET_DIR)" "$(BASE_TARGET_DIR)"; do \
 		[ -n "$$root" ] || continue; \
diff --git a/package/ingenic-sdk/ingenic-sdk.mk b/package/ingenic-sdk/ingenic-sdk.mk
index 8ab29367..c159f0e6 100644
--- a/package/ingenic-sdk/ingenic-sdk.mk
+++ b/package/ingenic-sdk/ingenic-sdk.mk
@@ -226,7 +226,7 @@ define INGENIC_SDK_INSTALL_TARGET_CMDS
 endef
 
 define INGENIC_SDK_INSTALL_TARGET_CMDS
-	krel="$$( $(MAKE) -s -C $(LINUX_DIR) kernelrelease 2>/dev/null )"; \
+	krel="$$( $(MAKE) --no-print-directory -s -C $(LINUX_DIR) kernelrelease 2>/dev/null )"; \
 	if [ -z "$$krel" ]; then krel="$(LINUX_VERSION_PROBED)"; fi; \
 	for root in "$(TARGET_DIR)" "$(BASE_TARGET_DIR)"; do \
 		[ -n "$$root" ] || continue; \
